### DISCUSSION TITLE
cube-studio的使用记载

### DISCUSSION THREAD

**Discussion by cyxnzb** [tencentmusic/cube-studio#38] [General]
大家可以分享一些**关于咱们平台**的一些**使用技巧**、**相关知识科普**等内容

**Comment by cyxnzb** [comment]
最初使用平台的时候遇到了一些问题。
因为咱们这个平台涉及面比较广，所以定位起来还是比较困难。
主要集中在**环境部署**上面，因此我先来分享几个给大家。

**Comment by cyxnzb** [comment]
我的话，采用的是**单机部署**的形式进行部署，机器是一台**t7920**，配置的话是标配
只不过显卡换成了RTX2070、**内存是64GB**，系统是Ubuntu18.04 基于Server的。

**Reply by cyxnzb** [reply]
在早期部署的时候遇到过以下几个问题：
①：**rancher无法安装作者所提到的2.5.2**
这个问题的解决比较玄学，早前安装的时候，装了10次左右，只有2次装上了2.5.2，其余的都是装上后，容器会自动重启，查日志也查不出来具体的，目前呢也没啥有效的解决办法，所以我后来都是拉的最新版安装
②：**集群创建后，部署很久**
这个问题，之前我是一直以为出错了，因为集群状态他会提示红色的警告文字，但是查看日志后，发现还是在运行，只不过时间比较久，**一般要十几分钟才能完全部署完**
③：**作者提供的notebook镜像中，GPU版有些组件缺少，如Tensorboard等，然后安装的话会出现部分问题**

**Reply by cyxnzb** [reply]
整个平台不拉notebook，内存占用大概10几G，256GB的硬盘占了84%

**Comment by cyxnzb** [comment]
然后单机部署的话，我个人拙见认为：
作者提供的**reset_docker.sh**文件，也就是cube-studio/tree/master/install/kubernetes/rancher/下的
对整个环境的清理**还不够彻底**，不过也**可能**这个文件**并不适合单机部署**。
然后我在这个文件的基础上自己编写了一个文件，**可供大家参考**

**Reply by cyxnzb** [reply]
#resetall.sh

#1.先把docker停了并重置
##删除所有
cd /home/ubuntu/cube-studio-master/install/kubernetes/rancher
sh reset_docker.sh
sh reset_docker.sh #避免没清理干净，再运行一遍

###ubuntu适用,清理kube包
apt remove -y kubectl
apt remove -y kubeadm


###清理一些k8s的东西
sudo rm -rvf $HOME/.kube
sudo rm -rvf ~/.kube/
sudo rm -rvf /etc/kubernetes/
sudo rm -rvf /etc/systemd/system/kubelet.service.d
sudo rm -rvf /etc/systemd/system/kubelet.service
sudo rm -rvf /usr/bin/kube*
sudo rm -rvf /etc/cni
sudo rm -rvf /opt/cni
sudo rm -rvf /var/lib/etcd
sudo rm -rvf /var/etcd
sudo rm -rvf /data/k8s
sudo rm -rvf /etc/rancher/
sudo rm -rf /usr/local/kubeflow-tools/

###清理docker的东西
docker system prune -fa #清理所有没用到的,无用的东西
docker rmi -f $(docker images -qa) #删除所有镜像,以重新部署

##清理项目的东西
#这个后续再补吧,因为如果作者更新的话,还得变
#主要清理kubernetes/config、kubeconfig/dev-kubeconfig(目录)

###ubuntu适用,清理docker
apt remove -y docker
apt remove -y docker-ce
sudo apt-get autoremove docker docker-ce docker-engine  docker.io  containerd runc
sudo apt-get autoremove docker-ce-*

#备份下配置文件
cp /etc/docker/daemon.json /home/ubuntu/


###linxu适用
yum remove docker-ce -y
yum remove -y docker
yum remove -y htop docker-compose
yum remove -y htop docker-composewireshark
yum remove -y wireshark

#reboot一下保证生效~
reboot

**Reply by cyxnzb** [reply]
下面这个是配套的安装sh文件：

**Reply by cyxnzb** [reply]
#installall.sh


#ubuntu刷新缓存
sudo apt-get -y autoclean 
sudo apt-get -y clean 
sudo apt-get -y autoremove
sudo apt-get -y update 

#ubuntu安装k8s组件
apt install -y kubectl
apt install -y kubeadm

#ubuntu安装docker
cat >/etc/apt/sources.list <<EOF
deb https://mirrors.cloud.tencent.com/ubuntu/ bionic main restricted universe multiverse
deb https://mirrors.cloud.tencent.com/ubuntu/ bionic-security main restricted universe multiverse
deb https://mirrors.cloud.tencent.com/ubuntu/ bionic-updates main restricted universe multiverse
EOF
apt update && apt-get install -y docker.io gcc make

#改动下docker
service docker stop
systemctl stop docker.service
systemctl stop docker.socket

#创建docker的存储路径，因为docker使用的存储很大，默认根目录磁盘很小
rm -rf /data/docker && mkdir -p /data/docker/

#ubuntu安装docker部件
apt install -y docker-ce

#避免服务在运行，影响部署，所以再运行一遍停止
systemctl stop docker.service
systemctl stop docker.socket

#将源docker目录下文件，复制到新目录下
cp -R /var/lib/docker/* /data/docker/

#将docker新目录写入到配置文件中
sed -i "s/containerd.sock/containerd.sock --graph \/data\/docker /g" /usr/lib/systemd/system/docker.service 

#将原有配置文件恢复
mkdir -p /etc/docker/
mv /home/ubuntu/daemon.json /etc/docker/daemon.json

#重载配置
systemctl daemon-reload

#重启docker
systemctl start docker 

#删除原目录数据
rm -rf /var/lib/docker

#reboot一下
reboot

**Reply by cyxnzb** [reply]
不过以上这些，还是希望大家根据自己的**实际环境**来编写修改以及执行，并且**注意数据的备份**，因为采取的是**直接删除**的形式！

**Comment by cyxnzb** [comment]
对了，不知道大家在配置config.yaml的时候，有没有遇到过**定位不准**的问题，比如说我想在第一行插入文字，结果插入到第二行去了（这里指的是**编辑配置**的模块）

**Comment by cyxnzb** [comment]
总结一下，单机部署是**可行的**，只不过需要注意的地方不少，作者提供的教程视频基本上也够用，能够完整的搭建平台并使用
[参考视频](https://www.bilibili.com/video/BV18r4y147oj/)

**Reply by ZXTFINAL** [reply]
部署时有遇到过 #77 这个问题吗，有没有解决方案，折腾了好久

**Reply by cyxnzb** [reply]
有遇到过类似的，但是我等了一会就好了

**Reply by ZXTFINAL** [reply]
> 有遇到过类似的，但是我等了一会就好了

那我再重新试下，多等会，周末没看见，sry : )

**Reply by ZXTFINAL** [reply]
> 

![image](https://user-images.githubusercontent.com/53009474/188355775-0665b335-f192-4ea7-8b4d-99634365b796.png)
等也不会好

**Comment by cyxnzb** [comment]
此外，因为个人的原因，对服务器做了个**内外网穿透**，也分享给大家一些经验：
我采取过两种方案，**①是花生壳 ②是frp**

**Reply by cyxnzb** [reply]
先来说说 **花生壳**
简单来说就是安装它的oray服务在服务器上，比较傻瓜式，就不赘述了
[下载地址](https://hsk.oray.com/download/#type=linux)
[网上教程](https://blog.csdn.net/forgetjoker/article/details/118163408)
有几个要注意的地方：
①**免费版够用**
实测免费版就够用了，只要映射把80映射出去就可以了，无需购买收费版**（访问量大除外）**
②**映射类型要选HTTP**
内网IP地址填写的话，单机部署是**装有cube-studio**平台的IP
比如我172.17.129.15装了，那么就填这个

**Reply by cyxnzb** [reply]
再来说说 **frp**
简单来说就是把frp-c （**客户端**）装在**部署平台**的机器上，frp-s（**服务端**）装在有**公网IP**等类似的机器上
[下载地址](https://github.com/fatedier/frp/releases) **版本其实我个人感觉任意**
有几个要注意的地方：
①**C/S用到的端口都要打开**
因为考虑到**端口映射**的问题，服务端客户端涉及到的端口都要对外开放，否则无法使用
②**建议配置密钥**
因为牵涉到cube-studio平台一些组件**能调用服务器资源**，最好**加上密钥**，避免恶意访问
③**建议配置开启自启动**
因为避免一些偶发情况，造成服务器意外重启，导致映射失效，所以**建议为frp配置开机自启**
[参考教程](https://blog.csdn.net/weixin_40959890/article/details/125202295)
注：宝塔面板 本人自测对frp的自启守护无效，不知道是不是个人原因，大家有兴趣可以测试下
④**牵涉到的域名建议备案**
例如.cn这些，要备案才能正常访问，所以为了规避麻烦，尽量提前备案或者直接采取.xyz等他方域名

**Comment by cyxnzb** [comment]
①**花生壳**穿透的访问速度取决于你选择的是**免费版**还是**收费版**以及是否开通了**加速包**一类的产品
②**frp**穿透的访问速度取决于你**服务端**（也就是有公网IP的机器）与**客户端**的访问速度，以及**域名DNS解析**，**CDN加速**这些
以上两种穿透的前提是**一定要能访问互联网**或者**有跳板机**，总之要想方设法将**部署平台的机器联网**才行

**Comment by cyxnzb** [comment]
如果有人跟我一样，喜欢直接用里面的Jupyter去执行一些代码、任务等等，那么我们可以尝试修改rancher里面部署cube-studio的机器里的configmap，然后将关于notebook续期时间的代码改为*30，这样可以30天续期一次，也可以*60，都行，记得重新部署kubeflow-dashboard,在工作负载里面
